### PR TITLE
#169523967 Integration test for infusion endpoints

### PR DESCRIPTION
**Pull request by okiti**
#### WHAT DOES THIS PR DO
- Implements more use cases for infusion endpoints in it's integration test

#### RELEVANT PT STORY
- [#169523967](https://www.pivotaltracker.com/story/show/169523967)

### CODE REVIEW

File: src/tests/infusion/deleteInfusion.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      headers: {
+        'req-token': context.testGlobals[WARD_USER].authToken,
+      },
+    }),
+    response: {
+      status: 204,
+      body: {},
+    },
+  },
+];
```
Comment (IAMOTZ) [inline src/tests/infusion/deleteInfusion.test.js]: How about other test cases such as if the user is not providing authentication creds, if the id of infusion to delete is not valid e.tc. We need tests for those.

File: src/tests/infusion/createInfusion.test.js [hunk 2]
```diff
@@ -43,6 +42,50 @@ const testCases = [
...
+    response: {
+      status: 403,
+      body: {
+        success: false,
+        message: 'You do not have access to this endpoint',
+      },
+    },
+  },
+  {
+    title: 'creation of infusion should fail if some of the required data such as patientName is missing',
```
Comment (IAMOTZ) [inline src/tests/infusion/createInfusion.test.js]: This test case is too broad, the fact that we get a correct validation response when `patientName` is missing does not guarantee that the same thing would happen when `startVolume` is missing so please add test case for all the possible validation.

Comment (okiti) [reply src/tests/infusion/createInfusion.test.js]: okay sir

File: src/tests/infusion/createInfusion.test.js [hunk 3]
```diff
@@ -43,6 +42,50 @@ const testCases = [
...
+      },
+    }),
+    response: {
+      status: 403,
+      body: {
+        success: false,
+        message: 'You do not have access to this endpoint',
+      },
+    },
+  },
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/tests/infusion/createInfusion.test.js]: fixed

File: src/tests/infusion/getInfusion.test.js [hunk 4]
```diff
@@ -100,6 +99,42 @@ const testCases = [
...
+  {
+    title: 'should fail to get infusion if the req-token is not part of the header',
+    request: context => ({
+      body: {},
+      path: '/api/infusion/555aa',
+      method: 'get',
+      headers: {
+        Auth: context.testGlobals[WARD_USER].authToken,
+      },
+    }),
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/tests/infusion/getInfusion.test.js]: Fixed

File: src/tests/infusion/createInfusion.test.js (hunk 2 above)
Comment (okiti) [reply src/tests/infusion/createInfusion.test.js]: Fixed

File: src/tests/infusion/updateInfusion.test.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }),
+    response: {
+      status: 404,
+      body: {
+        success: false,
+        message: 'Infusion not found',
+      },
+    },
+  },
+];
```
[the remark this answers is not in the document]
Comment (okiti) [reply src/tests/infusion/updateInfusion.test.js]: Fixed

File: src/tests/infusion/deleteInfusion.test.js (hunk 1 above)
Comment (okiti) [reply src/tests/infusion/deleteInfusion.test.js]: fixed

**Pull request merged**